ports/stm32/rfcore.c

```diff
@@ -605,4 +618,67 @@ void IPCC_C1_RX_IRQHandler(void) {
     IRQ_EXIT(IPCC_C1_RX_IRQn);
 }
 
+// AN5185
+#define MAGIC_FUS_ACTIVE 0xA94656B9
+// AN5289
+#define MAGIC_IPCC_MEM_INCORRECT 0x3DE96F61
+
+STATIC mp_obj_t rfcore_fus_active(void) {
+    if (ipcc_mem_dev_info_tab.fus.table_state == MAGIC_IPCC_MEM_INCORRECT) {
+        mp_raise_OSError(MP_EINVAL);
+    }
+    return mp_obj_new_bool(ipcc_mem_dev_info_tab.fus.table_state == MAGIC_FUS_ACTIVE);
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_0(rfcore_fus_active_obj, rfcore_fus_active);
+
+STATIC mp_obj_t get_version_tuple(uint32_t data) {
+    mp_obj_t items[] = {
+        MP_OBJ_NEW_SMALL_INT(data >> 24), MP_OBJ_NEW_SMALL_INT(data >> 16 & 0xFF), MP_OBJ_NEW_SMALL_INT(data >> 8 & 0xFF), MP_OBJ_NEW_SMALL_INT(data >> 4 & 0xF), MP_OBJ_NEW_SMALL_INT(data & 0xF)
+    };
+    return mp_obj_new_tuple(5, items);
+}
+
+STATIC mp_obj_t rfcore_fus_version(void) {
+    uint32_t v = (rfcore_fus_active() == mp_const_true) ? ipcc_mem_dev_info_tab.fus.fus_version : ipcc_mem_dev_info_tab.ws.fus_version;
+    return get_version_tuple(v);
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_0(rfcore_fus_version_obj, rfcore_fus_version);
+
+STATIC mp_obj_t rfcore_ws_version(void) {
+    uint32_t v = (rfcore_fus_active() == mp_const_true) ? ipcc_mem_dev_info_tab.fus.ws_version : ipcc_mem_dev_info_tab.ws.fw_version;
+    return get_version_tuple(v);
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_0(rfcore_ws_version_obj, rfcore_ws_version);
+
+STATIC mp_obj_t rfcore_sys_hci(mp_obj_t ogf_in, mp_obj_t ocf_in, mp_obj_t cmd_in) {
+    rfcore_fus_active(); // Fail if mem layout incorrect.
+    mp_int_t ogf = mp_obj_get_int(ogf_in);
+    mp_int_t ocf = mp_obj_get_int(ocf_in);
+    mp_buffer_info_t bufinfo = {0};
+    mp_get_buffer_raise(cmd_in, &bufinfo, MP_BUFFER_READ);
+    ssize_t len = tl_sys_hci_cmd_resp(HCI_OPCODE(ogf, ocf), bufinfo.buf, bufinfo.len);
+    if (len < 0) {
+        mp_raise_OSError(-len);
+    }
+    return mp_obj_new_bytes(ipcc_membuf_sys_cmd_buf, len);
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_3(rfcore_sys_hci_obj, rfcore_sys_hci);
+
+STATIC const mp_rom_map_elem_t rfcore_module_globals_table[] = {
+    { MP_ROM_QSTR(MP_QSTR___name__), MP_ROM_QSTR(MP_QSTR_rfcore) },
+
+    { MP_ROM_QSTR(MP_QSTR_fus_active), MP_ROM_PTR(&rfcore_fus_active_obj) },
+    { MP_ROM_QSTR(MP_QSTR_fus_version), MP_ROM_PTR(&rfcore_fus_version_obj) },
+    { MP_ROM_QSTR(MP_QSTR_ws_version), MP_ROM_PTR(&rfcore_ws_version_obj) },
+
+    { MP_ROM_QSTR(MP_QSTR_sys_hci), MP_ROM_PTR(&rfcore_sys_hci_obj) },
+};
```

Comment:
These functions could go in the `stm` module... defining a new `rfcore` module does seem logical and self-contained, but prevents the name "rfcore" from being used in an application for other things.  Don't know if that'd be an issue, just a point for discussion.